columnflow/production/cms/gen_top_decay.py

```diff
@@ -13,7 +13,7 @@
 
 
 @producer(
-    uses={"nGenPart", "GenPart.genPartIdxMother", "GenPart.pdgId", "GenPart.statusFlags"},
+    uses={"GenPart.genPartIdxMother", "GenPart.pdgId", "GenPart.statusFlags"},
     produces={"gen_top_decay"},
```

Comment:
This is a good point. I don't have this producer in the analysis workflow that I was using for testing, but it would indeed be important to check that this doesn't break things. Since the producer does not explicitly require `nGenPart` for the computation, I think it should be safe to remove it. If you are using it in your analysis, do you think you could run a quick test to confirm?

Reply:
I did not try this specific producer, but for my signal-specific Gen producer, the `"nGenPart"` column seems to be needed (`AttributeError: no field named 'genPartIdxMotherG'`, even though I added `GenPart.*` to my `uses`).

So to be safe, we might just want to keep `nGenPart` in there and skip the check for this producer?

Reply:
Ah, I see. I assume you mean the [`gen_hbw_decay_products`](https://github.com/uhh-cms/hh2bbww/blob/3165e824c63dadeaf8bf930b58a99844efbeefc1/hbw/production/gen_hbw_decay.py#L17) producer? In that case the `nGenPart` information is likely needed by the NanoAOD schema for the `distinctParent` accessor to work correctly.

For the `gen_top_decay` producer, the `nGenPart` would also be needed for  `distinctChildrenDeep`, but I find that adding that to the columns when loading a NanoAOD file with `coffea` leads to the error in the PR description. For that reason, `nGenPart` is the only column that is not being added automatically for NanoAOD.

However, I am a bit surprised that your producer runs at all if you *do* add `nGenPart` to the `uses`, as for me the issue occurs even before the producer code is executed. But I see now that there was an update from @riga to the software sandboxes, so maybe the issue was fixed upstream and we can include `nGenPart` back. I'll do some testing.

Reply:
> To deal with the entry-numbering columns n*, which are used by coffea to reconstruct the structure of jagged columns but are not present as columns in the AwkwardArray itself, the list of filters used when constructing the event chunks with coffea.nanoevents.NanoEventsFactory.from_root is extended by an n-prefixed version of all routes present in the file (a superset of all possible entry-numbering columns). This eliminates the need to specify these columns manually under uses, at the expense of loading potentially unrelated columns starting with n, which seems very unlikely to happen to begin with.

Maybe we could offload this to the IO? I.e., in case one or more columns under "Jet" are to be loaded AND the input format is ROOT, the "nJet" column is loaded as well. This way, when a parquet file (with already existing starts / stops info) is read, no extra counter column is loaded.

Reply:
You're right of course, I think I confused two PRs :)